agent/engine/execcmd/manager.go

```diff
@@ -115,3 +116,12 @@ func IsExecEnabledContainer(container *apicontainer.Container) bool {
 	_, ok := container.GetManagedAgentByName(ExecuteCommandAgentName)
 	return ok
 }
+
+// HasExecAgentInitFailed returns if exec feature has been disabled for a container
+// because of exec agent initialization failure
+func HasExecAgentInitFailed(ma apicontainer.ManagedAgent) (string, bool) {
+	if ma.ID == "" && ma.Status == apicontainerstatus.ManagedAgentStopped {
```

Comment:
this is probably not the right check. If restart fails after many retries we move the status to stopped, which this method returns true for, but we may want to restart again. 
one option here could be to introduce a bool flag to explicitly indicate exec init has failed.

Reply:
Thought about that. If the the ID is "", then the exec agent is not initialized. I am checking that as well. It is only after successful initialization, we will give an uuid as its id. That is why a "" Id will indicate that exec agent is not initialized. Let me know if you think otherwise

Reply:
ah, got it. Could you add this as a comment? (in case this logic changes in the future)

Reply:
sure